eg_derive/src/lib.rs

```diff
@@ -87,7 +87,24 @@ fn example(data: Data) -> TokenStream {
                 }
             }
         }
-        Data::Union(_) => unimplemented!(),
+        Data::Union(_union) =>  {
+            let mut tagged_fields = _union.fields.named.iter().filter(|field| {
+                field.attrs.iter().any(|attr| {
+                    matches!(attr.style, AttrStyle::Outer)
+                        && matches!(attr.path.get_ident(), Some(ident) if ident == "eg")
+                })
+            });
+            let eg_field = match tagged_fields.next() {
+                None => panic!("A union must have one field tagged with `eg`."),
+                Some(tagged_variant) => tagged_variant,
+            };
+            if tagged_fields.next().is_some() {
+                panic!("Cannot tag more than one union field with `eg`.");
+            }
+            let field_eg = eg_field.ident.as_ref().unwrap(); // unnamed fields only apply to tuple struct definitions
+            let field_type = &eg_field.ty;
+            quote!(Self { #field_eg: #field_type })
```

Comment:
Here's the problematic line; the field being created was being passed a type instead of a value and the compiler was (understandably) confused! This is a relatively simple fix, so you can push it through or I can pick it up from here; let me know what you'd prefer 👍 

```suggestion
            quote!(Self { #field_eg: #field_type::eg() })
```

Reply:
:facepalm: Nice, completely missed that! The error message was so confusing

Reply:
I did the same exact thing, but I think I just memory holed the actual macro expansion bit haha, I glanced at it and didn't even realise it was wrong 🤦